### ISSUE TITLE
Upgrading to v5.1 raises FileNotFoundError on `ubuntu 20.04` and `gcc-9`

### ISSUE THREAD

**Issue by ghorbanzade** [gcovr/gcovr#633]
Thank you for your work on this project!

**Describe the bug**
We use gcovr in GitHub Actions of our open-source project [trytouca/trytouca](https://github.com/trytouca/trytouca) to generate coverage report for our C++ SDK. Upgrading gcovr to v5.1 consistently breaks our CI build with the following message:

```
FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/trytouca/trytouca/sdk/cpp/stl_pair.h##c8fb74ae159354e1570195788e158636.gcov'
```

See our [GitHub Actions log](https://github.com/trytouca/trytouca/runs/6940925774?check_suite_focus=true) (raw format [here](https://pipelines.actions.githubusercontent.com/serviceHosts/da02a9ba-e760-4848-9ac1-ee6591b03101/_apis/pipelines/1/runs/337/signedlogcontent/2?urlExpires=2022-06-17T20%3A31%3A05.0761976Z&urlSigningMethod=HMACV1&urlSignature=tvge1ICzFeROMHsLnvS7AESnncm8QKIQl%2F7rDEKoUnM%3D)) for context.

**To Reproduce**
Our CI workflow is available [here](https://github.com/trytouca/trytouca/blob/main/.github/workflows/build.yml#L212-L230) and very straight-forward. As a temporarily issue to avoid a broken build, we've pinned gcovr to v5.0. But we'd like to upgrade to the new version.

**Expected behavior**
We think the exception being thrown does not make sense. So the build should succeed just like it used to in gcovr 5.0.

**Desktop (please complete the following information):**
 - OS: ubuntu-20.04
 - GCC version: GNU 9.4.0
 - GCOVR version: 5.1

**Additional context**
- Gcovr 5.1 works just fine with this same project, using macoS Montery with any of the clang or gcc-9 compilers.

**Comment by Spacetown** [comment]
Hi, this is already solved on master, see #587.
@latk I think we should finish the open Pars and create a 5.2 now.

**Comment by ghorbanzade** [comment]
Thanks @Spacetown for the prompt response 🙌🏼. Good to hear.
